### PR TITLE
fix: rendering of gender field

### LINKED ISSUE #3307: Custom Form : Buggy Fields in custom forms
**Describe the bug**

Some of the fields of custom forms are misbehaving when setting to required.
**To Reproduce**
Steps to reproduce the behaviour:
1. Create an event with `Gender Field` of `Attendee Form` as a required field.
2. Try to place an order of the following event
3. Scroll down to 'Gender on Order Page'

**Expected behaviour**

Remove the bugs
**Screenshots**

image

**Desktop (please complete the following information):**
 - OS: Mac
 - Browser Chrome
 - Version 74.3.0

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #3307 

#### Short description of what this resolves:

- Fixing the rendering of `if` clause for gender fields

#### Screenshot 

image

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/templates/components/forms/orders/order-form.hbs [hunk 1]
```diff
@@ -57,24 +57,26 @@
...
                 {{/if}}
               {{/if}}
+              {{#if (eq field.fieldIdentifier 'gender')}}
+                {{#ui-dropdown class='search selection' value=(mut (get holder field.fieldIdentifier)) onChange=(action (mut holder.gender)) as |execute mapper|}}
+                  <i class="dropdown icon"></i>
+                  <div class="default text">{{t 'Select your gender'}}</div>
+                  <div class="menu">
+                    {{#each genders as |gender|}}
+                      <div class="item" data-value="{{map-value mapper gender.name}}">
+                        {{gender.name}}
```
Comment (mrsaicharan1) [inline app/templates/components/forms/orders/order-form.hbs]: translation helpers

Comment (kushthedude) [reply app/templates/components/forms/orders/order-form.hbs]: won't work, Its not plain text.

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 @uds5501 @shreyanshdwivedi Please REview

**Comment by kushthedude** [comment]
No need for server change @mrsaicharan1 only issue was it was placed in
wrong parent if block for text type and on server it was changed to select .

**Comment by mrsaicharan1** [comment]
> No need for server change @mrsaicharan1 only issue was it was placed in wrong parent if block for text type and on server it was changed to select .
> […](#)

I was asking for that very change. 👍

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 @uds5501 @shreyanshdwivedi @mrsaicharan1 Please REview

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Please Review

**Comment by kushthedude** [comment]
@uds5501 Please make sure you do `Squash and Merge` so that multiple commits don't associate with a single PR. 😅

**Pull request merged**